src/main/java/slimeknights/tconstruct/library/materials/json/MaterialStatJsonWrapper.java

```diff
@@ -22,6 +25,11 @@ public MaterialStatJsonWrapper(List<BaseMaterialStatsJson> stats) {
     return stats;
   }
 
+  @Nullable
+  public MaterialId getMaterialId() {
+    return materialId != null ? new MaterialId(materialId) : null;
```

Comment:
Why not convert this in the constructor so we do not keep throwing away MaterialId objects every time this is called?

Reply:
Because deserialization does not use the constructor, and I was too lazy to write a special deserializer for it.